runtime/src/snapshot_minimizer.rs

```diff
@@ -0,0 +1,609 @@
+use {
+    crate::{
+        accounts_db::{AccountStorageEntry, AccountsDb, PurgeStats},
+        bank::Bank,
+    },
+    dashmap::DashSet,
+    log::info,
+    rayon::{
+        iter::{IntoParallelIterator, IntoParallelRefIterator, ParallelIterator},
+        prelude::ParallelSlice,
+    },
+    solana_measure::measure,
+    solana_sdk::{
+        account::ReadableAccount,
+        account_utils::StateMut,
+        bpf_loader_upgradeable::{self, UpgradeableLoaderState},
+        clock::Slot,
+        pubkey::Pubkey,
+    },
+    std::{
+        collections::HashSet,
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc, Mutex,
+        },
+    },
+};
+
+pub struct SnapshotMinimizer {
+    bank: Arc<Bank>,
+    starting_slot: Slot,
+    ending_slot: Slot,
+    minimized_account_set: DashSet<Pubkey>,
+}
+
+impl SnapshotMinimizer {
+    /// Removes all accounts not necessary for replaying slots in the range [starting_slot, ending_slot].
```

Comment:
I think the side effects of this function are worth mentioniong. It apears to modify the bank and various data structures, cap, etc.

Reply:
that's the point of this function - accumulate and then modify the bank and accounts db, right?